files/k3s-install-agent.sh

```diff
@@ -12,32 +12,76 @@ do
 done
 }
 
-# Disable firewall 
-/usr/sbin/netfilter-persistent stop
-/usr/sbin/netfilter-persistent flush
-
-systemctl stop netfilter-persistent.service
-systemctl disable netfilter-persistent.service
-
-# END Disable firewall
+if test -f /etc/lsb-release; then
+  operating_system="ubuntu"
+else
+  operating_system="oraclelinux"
+fi
+
+if [[ "$operating_system" == "ubuntu" ]]; then
+  # Disable firewall 
+  /usr/sbin/netfilter-persistent stop
+  /usr/sbin/netfilter-persistent flush
+
+  systemctl stop netfilter-persistent.service
+  systemctl disable netfilter-persistent.service
+
+  # END Disable firewall
+
+  apt-get update
+  apt-get install -y software-properties-common jq
+  DEBIAN_FRONTEND=noninteractive apt-get upgrade -y
+  %{ if install_nginx_ingress }
+  DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y python3 python3-pip nginx
+  systemctl enable nginx
+  pip install oci-cli
+  %{ endif }
+fi
+
+if [[ "$operating_system" == "oraclelinux" ]]; then
+  # Disable firewall
+  systemctl disable --now firewalld
+  # END Disable firewall
+
+  # Fix iptables/SELinux bug
+  echo '(allow iptables_t cgroup_t (dir (ioctl)))' > /root/local_iptables.cil
+  semodule -i /root/local_iptables.cil
+
+  dnf -y update
+  dnf -y install jq curl
+  %{ if install_nginx_ingress }
+  if grep -q "el9" /etc/os-release; then
+    dnf -y install python39-oci-cli python3-jinja2 nginx-all-modules
+  else
+    dnf -y module enable nginx:1.20 python36:3.6
+    dnf -y install python36-oci-cli python3-jinja2 nginx-all-modules
+  fi
+  %{ endif }
 
-apt-get update
-apt-get install -y software-properties-common jq
-DEBIAN_FRONTEND=noninteractive apt-get upgrade -y
-DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y python3 python3-pip nginx
-systemctl enable nginx
-pip install oci-cli
+  # Nginx Selinux Fix
+  setsebool httpd_can_network_connect on -P
+fi
 
 local_ip=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/vnics/ | jq -r '.[0].privateIp')
 flannel_iface=$(ip -4 route ls | grep default | grep -Po '(?<=dev )(\S+)')
 
+k3s_install_params=("--node-ip $local_ip")
+k3s_install_params+=("--flannel-iface $flannel_iface")
+
+if [[ "$operating_system" == "oraclelinux" ]]; then
+  k3s_install_params+=(="--selinux")
```

Comment:
```suggestion
  k3s_install_params+=("--selinux")
```

Reply:
Fixed, in both k3s-install-agent.sh and k3s-install-server.sh